src/components/Modals/Relationship/StepOne.jsx

```diff
@@ -1,24 +1,107 @@
+/* eslint-disable jsx-a11y/no-noninteractive-element-interactions */
+/* eslint-disable jsx-a11y/click-events-have-key-events */
+import { useState, useContext } from "react";
+import { useQuery } from "@tanstack/react-query";
+
+import fetchData from "../../../utils/axios";
+
+import UserContext from "../../../context/UserContext";
+import CurrentDBIdContext from "../../../context/CurrentDBIdContext";
 import Content from "../SharedItems/Content";
 import Title from "../SharedItems/Title";
 import Button from "../../shared/Button";
 import Message from "../SharedItems/Message";
+import Loading from "../../shared/Loading";
+
+function StepOne({
+  setRelationshipStep,
+  databaseName,
+  relationData,
+  setRelationData,
+}) {
+  const [targetDatabases, setTargetDatabases] = useState([]);
+  const [isSelected, setIsSelected] = useState("");
+
+  const { userId } = useContext(UserContext);
+  const currentDBId = useContext(CurrentDBIdContext);
+
+  async function getDatabaseList() {
+    const response = await fetchData("GET", `users/${userId}/databases`);
+
+    return response.data.databases;
+  }
+
+  const { isLoading } = useQuery(["dbs"], getDatabaseList, {
+    enabled: !!userId,
+    onSuccess: result => {
+      const filteredDbs = result.filter(
+        database => database._id !== currentDBId,
+      );
+
+      setTargetDatabases(filteredDbs);
+    },
+    onFailure: () => {
+      console.log("sending user to errorpage");
+    },
+    refetchOnWindowFocus: false,
+  });
+
+  if (isLoading) {
+    return <Loading />;
+  }
+
+  function handleNextClick() {
+    if (!relationData.foreignDbId) {
+      alert("Please choose database to make relationship!");
+
+      return;
+    }
+
+    setRelationshipStep("stepTwo");
+  }
+
+  function handleDatabaseClick(id) {
+    setIsSelected(id);
+
+    setRelationData({
+      ...relationData,
+      foreignDbId: id,
+    });
+  }
 
-function StepOne({ setRelationshipStep }) {
   return (
     <>
       <Title>Step 1</Title>
       <Message>
         <p>Please choose a database that you would like to link with DBNAME</p>
       </Message>
       <Content>
-        <div className="flex justify-center items-center h-60 w-60">
-          database card here
+        <div className="flex flex-col justify-around items-center h-auto w-60">
```

Comment:
CSS 속성값의 순서를 맞춰주시면 이후 css 변경작업이 이루어져야할 경우 한결 편할 것 같습니다!

```suggestion
        <div className="flex flex-col justify-around items-center w-60 h-auto ">
```

Reply:
네~ 반영하겠습니다~